src/ActionList/Item.tsx

```diff
@@ -228,12 +265,29 @@ export const Item = React.forwardRef<HTMLLIElement, ActionListItemProps>(
         >
           <ItemWrapper {...wrapperProps}>
             <Selection selected={selected} />
-            {slots.leadingVisual}
+            {
```

Comment:
By the way, I cleaned up all this logic while I was working on the [loading state](https://github.com/github/primer/issues/2681). I put it in a component, made the logic a lot easier to follow, and was able to get rid of the non-null assertion.

If you'd like me to introduce that component in this PR instead of the loading state PR (not yet created), I'm ok with that.